Client/mods/deathmatch/logic/CClientColPolygon.cpp

```diff
@@ -69,20 +69,68 @@ void CClientColPolygon::SetPosition(const CVector& vecPosition)
     // Add queued collider refresh for v1.1
 }
 
-void CClientColPolygon::AddPoint(CVector2D vecPoint)
+void CClientColPolygon::AddPoint(CVector2D vecPoint, int iPointIndex)
 {
-    float fDistanceX = vecPoint.fX - m_vecPosition.fX;
-    float fDistanceY = vecPoint.fY - m_vecPosition.fY;
+    if (iPointIndex < 0)
+    {
+        m_Points.push_back(vecPoint);
+    }
+    else
+    {
+        if (iPointIndex >= m_Points.size())
+            return;
```

Comment:
what do you think about returning false here (and therefore also the lua function)?